scipy/integrate/_ivp/tests/test_ivp.py

```diff
@@ -1085,9 +1085,116 @@ def fun_with_arg(t, y, a):
 def test_initial_state_finiteness(f0_fill):
     # regression test for gh-17846
     msg = "All components of the initial state `y0` must be finite."
     with pytest.raises(ValueError, match=msg):
         solve_ivp(fun_zero, [0, 10], np.full(3, f0_fill))
+
+
+@pytest.mark.parametrize('method', ['RK23', 'RK45', 'DOP853', 'Radau', 'BDF'])
+def test_zero_interval(method):
+    # Case where upper and lower limits of integration are the same
+    # f[y(t)] = 2y(t)
+    def f(t, y):
+        return 2 * y
+    res = solve_ivp(f, (0., 0.), np.array([1]), method = method)
+    assert_(res.success)
+    assert_equal(res.y, 0.)
+    
+
+@pytest.mark.parametrize('method', ['RK23', 'RK45', 'DOP853', 'Radau', 'BDF'])
+def test_tbound_respected_small_interval(method):
+    """Regression test for gh-17341"""
+    SMALL = 1e-4
+
+    # f[y(t)] = 2y(t) on t in [0,SMALL]
+    #           undefined otherwise 
+    def f(t, y):
+        if t > SMALL:
+            raise ValueError("Function was evaluated outside interval")
+        return 2 * y
+    res = solve_ivp(f, (0., SMALL), np.array([1]), method=method)
+    assert_(res.success)
+
+
+@pytest.mark.parametrize('method', ['RK23', 'RK45', 'DOP853', 'Radau', 'BDF'])
+def test_tbound_respected_larger_interval(method):
+    """Regression test for gh-8848"""
+    def V(r):
+        return -11/r + 10 * r / (0.05 + r**2)
+
+    def func(t, p):
+        if t < -17 or t > 2:
+            raise ValueError("Function was evaluated outside interval")
+        P = p[0]
+        Q = p[1]
+        r = np.exp(t)
+        dPdr = r * Q
+        dQdr = -2.0 * r * ((-0.2 - V(r)) * P + 1 / r * Q)
+        return np.array([dPdr, dQdr])
+
+    result = solve_ivp(func, 
+                       (-17, 2),
+                       y0 = np.array([1, -11]),
+                       max_step = 0.03,
+                       vectorized = False,
+                       t_eval = None,
+                       atol = 1e-8, 
+                       rtol = 1e-5)
+    assert_(result.success)
+
+
+@pytest.mark.parametrize('method', ['RK23', 'RK45', 'DOP853', 'Radau', 'BDF'])
+def test_tbound_respected_oscillator(method):
+    "Regression test for gh-9198"
+    def reactions_func(t, y):
+        if (t > 205): 
+            raise ValueError("Called outside interval")
+        yprime = np.array([1.73307544e-02, 
+                           6.49376470e-06, 
+                           0.00000000e+00, 
+                           0.00000000e+00])
+        return yprime
+
+    def run_sim2(t_end, n_timepoints=10, shortest_delay_line=10000000):
+        init_state = np.array([134.08298555, 138.82348612, 100., 0.])
+        t0 = 100.0
+        t1 = 200.0
+        return solve_ivp(reactions_func,
+                         (t0, t1),
+                         init_state.copy(), 
+                         dense_output = True, 
+                         max_step = t1 - t0)
+    result = run_sim2(1000, 100, 100)
+    assert_(result.success)
```

Comment:
Here and in the other new tests added we now favour plain asserts
```suggestion
    assert result.success
```